File: federalism/fed-frontend/scripts/test_signed_url.mjs
import axios from 'axios';

const API_BASE = process.argv[2] || 'http://localhost:8000';
const MEDIA_ID = process.argv[3] || '1';

async function main() {
  const url = `${API_BASE}/api/media_manager/media_file/${MEDIA_ID}/signed_url/`;
  console.log(`Requesting signed URL from ${url}`);

  try {
    const response = await axios.get(url, { headers: { Accept: 'application/json' } });
    if (!response.data?.signed_url) {
      console.error('No signed_url found in response:', response.data);
      process.exit(1);
    }
    console.log('Signed URL received:', response.data.signed_url);

    const signedUrlResponse = await axios.get(response.data.signed_url, {
      responseType: 'stream',
      headers: { 'User-Agent': 'fed-frontend-signed-url-test/1.0' },
      validateStatus: (status) => status < 500,
    });

    console.log('Signed URL request status:', signedUrlResponse.status);
    if (signedUrlResponse.status !== 200) {
      console.error('Signed URL request failed');
      process.exit(1);
    }

    console.log('Frontend signed URL smoke test passed.');
  } catch (error) {
    console.error('Error testing signed URL:', error.message || error);
    process.exit(1);
  }
}

main();

File: federalism/scripts/test_signed_url.py
#!/usr/bin/env python3
import json
import sys
from urllib.request import urlopen, Request
from urllib.error import URLError, HTTPError


def main(api_base='http://localhost:8000'):
    media_id = 1
    url = f'{api_base}/api/media_manager/media_file/{media_id}/signed_url/'
    print(f'Checking signed URL endpoint: {url}')

    try:
        req = Request(url, headers={'Accept': 'application/json'})
        with urlopen(req) as resp:
            data = json.load(resp)
    except HTTPError as e:
        print('HTTP error:', e.code, e.reason)
        sys.exit(1)
    except URLError as e:
        print('URL error:', e.reason)
        sys.exit(1)

    if 'signed_url' not in data:
        print('Failure: signed_url missing in response:', data)
        sys.exit(1)

    signed_url = data['signed_url']
    print('Signed URL received:', signed_url)

    try:
        req = Request(signed_url, headers={'User-Agent': 'fed-educ-test/1.0'})
        with urlopen(req) as resp:
            print('Signed URL HEAD request succeeded:', resp.status)
    except HTTPError as e:
        print('Signed URL HTTP error:', e.code, e.reason)
        sys.exit(1)
    except URLError as e:
        print('Signed URL error:', e.reason)
        sys.exit(1)

    print('Signed URL endpoint test passed.')


if __name__ == '__main__':
    base = sys.argv[1] if len(sys.argv) > 1 else 'http://localhost:8000'
    main(base)
